### PR TITLE
fix an error when fp16 is used

### PR DESCRIPTION
**Pull request by Elin24**
two points:
1. the device is not required since `to` will complete all related things (shape & device);
2. When I use half type (fp16), line 56 meets errors since the type of `attn. sum()` is float32, I'm not sure why this happen, but the method `to` of Tensor could fix it no matter whether `float32` or `float16` is used.

### CODE REVIEW

**Comment by Elin24** [comment]
You can try this script:
```python
import torch
from slot_attention import SlotAttention

slot_attn = SlotAttention(
        num_slots = 5,
        dim = 512,
        iters = 3
    ).half().cuda()

    inputs = torch.randn(2, 1024, 512).half().cuda()
    out = slot_attn(inputs)

    print(out.dtype)
```
I apply `slot attention` to an old model using `apex.amp` instead of newest `autocast`, which meets the RuntimeError.

**Comment by Elin24** [comment]
It meets the same problem. You can have a try like this:
```python
from apex import amp

slot_attn = SlotAttention(
        num_slots = 5,
        dim = 512,
        iters = 3
    ).cuda()

    slot_attn = amp.initialize(slot_attn, opt_level='O1')

    inputs = torch.randn(2, 1024, 512).half().cuda()
    out = slot_attn(inputs)

    print(out.dtype)
```

`attn` is fp16, but `attn.sum(dim=-1, keepdim=True)` is fp32, which is wired. Thus I use `to(attn)` to convert the latter to fp16, but I'm not sure whether the operator `to` will detach this tensor.

**Comment by lucidrains** [comment]
@Elin24 i think nvidia apex is deprecated now https://github.com/NVIDIA/apex#1-amp--automatic-mixed-precision you should be using the native amp from pytorch

**Comment by lucidrains** [comment]
@Elin24 it works fine for me using pytorch amp

**Comment by Elin24** [comment]
@lucidrains yes, you are right. Thanks for your reply.
